java/client/build.gradle

```diff
@@ -250,7 +251,52 @@ tasks.withType(Test) {
     }
     // This is needed for the FFI tests
     jvmArgs "-Djava.library.path=${projectDir}/../target/release"
+
+    finalizedBy jacocoTestReport, jacocoTestCoverageVerification
+}
+
+// JaCoCo section (code coverage by unit tests)
+jacoco {
+    toolVersion = "0.8.9"
+}
+jacocoTestReport {
+    reports {
+        xml.configure { enabled false }
+        csv.configure { enabled false }
+        html.configure { enabled true } 
+    }
+
+    afterEvaluate {
+        classDirectories.setFrom(files(classDirectories.files.collect {
+            fileTree(dir: it)
+        }))
+    }
 }
+jacocoTestCoverageVerification {
+    violationRules {
+        rule {
+            element = 'CLASS'
+            excludes = [
```

Comment:
Add here protobuf generated files and netty related classes - we cannot cover them by UT